### PR TITLE
[Heartbeat Logging] Migrate Installations from using `FIRHeartbeatInfo` to `FIRHeartbeatLogger`

### PR DESCRIPTION
**Pull request by ncooke3**
#no-changelog

### CODE REVIEW

File: .github/workflows/installations.yml [hunk 1]
```diff
@@ -22,6 +22,8 @@ jobs:
     runs-on: macos-11
     env:
       plist_secret: ${{ secrets.GHASecretsGPGPassphrase1 }}
+      # TODO(v9): Remove `POD_LIB_LINT_ONLY` after publishing `HeartbeatLoggingTestUtils`.
```
Comment (paulb777) [inline .github/workflows/installations.yml:25]: I'm not sure we want to publish `HeartbeatLoggingTestUtils`. The test specs don't need to be runnable from `pod spec lint`.

**Review by paulb777 — APPROVED** [review]
What is the integration test plan?

File: .github/workflows/installations.yml (hunk 1 above)
Comment (ncooke3) [reply .github/workflows/installations.yml:25]: Ahh I see, thanks. So `HeartbeatLoggingTestUtils` would only be linted on CI when we're pulling in local sources. 

I added this TODO in a bunch of places so I can go back and revert those TODOs. cc: @ryanwilson

Comment (paulb777) [reply .github/workflows/installations.yml:25]: Right. SGTM

Comment (ryanwilson) [reply .github/workflows/installations.yml:25]: Sorry for not catching that, thanks!

**Comment by ncooke3** [comment]
> What is the integration test plan?

@paulb777, thanks for calling that out. This was actually quite tricky, but I added verification that the heartbeat integration is working as expected in [a40e06a](https://github.com/firebase/firebase-ios-sdk/pull/9543/commits/a40e06ad540774f21751935aa0891c9c50156caa).

File: .github/workflows/installations.yml (hunk 1 above)
Comment (ncooke3) [reply .github/workflows/installations.yml:25]: Cleaned these up in https://github.com/firebase/firebase-ios-sdk/pull/9554

**Review by paulb777 — APPROVED** [review]
Thanks!

**Pull request merged**